### PR TITLE
Update base image for a release 4

### PR DESCRIPTION
**Pull request by gberaudo**
Todo:
- [x] discuss with @sbrunner what we should do lor release 4 and what can be done later
- [x] -stop publishing the full and standard images (discuss it with GS devs);- (to discuss)
- [x] introduce an internal build in Docker to speed up development (of c2cwsgiutils itself)
- [x] remove fakepython
- [ ] push to master so that we can test

### CODE REVIEW

**Comment by sbrunner** [comment]
List of changes that should be done for release 4 ...
#530

File: .dependabot/config.yaml [hunk 1]
```diff
@@ -17,7 +17,7 @@ update_configs:
   - package_manager: python
     directory: /
     update_schedule: live
-    target_branch: release_3
+    target_branch: release_4
```
Comment (sbrunner) [inline .dependabot/config.yaml:20]: The security update should also be done on branch 3...

File: Dockerfile [hunk 2]
```diff
@@ -1,21 +1,19 @@
...
         gnupg \
         $DEV_PACKAGES && \
     equivs-build /opt/c2cwsgiutils/fake_python3 && \
-    dpkg -i python3_3.7.1-1~18.04_amd64.deb && \
-    rm python3_3.7.1-1~18.04_amd64.deb && \
     ln -s pip3 /usr/bin/pip && \
-    ln -s python3.7 /usr/bin/python && \
-    ln -sf python3.7 /usr/bin/python3 && \
+    ln -s python3.8 /usr/bin/python && \
+    ln -sf python3.8 /usr/bin/python3 && \
```
Comment (sbrunner) [inline Dockerfile:16]: Shouldn't be needed ...

Comment (gberaudo) [reply Dockerfile]: Tests were broken without it. See commit comment.

File: c2cwsgiutils/scripts/genversion.py [hunk 3]
```diff
@@ -11,7 +11,12 @@
 def _get_packages_version():
     result = {}
     with open(os.devnull, "w") as devnull:
-        for comp in subprocess.check_output(["pip3", "freeze"], stderr=devnull).decode().strip().split("\n"):
+        for comp in (
+            subprocess.check_output(["python3", "-m" "pip", "freeze"], stderr=devnull)
```
Comment (arnaud-morvan) [inline c2cwsgiutils/scripts/genversion.py:15]: Missing comma between "-m" and "pip" ?

Comment (fredj) [reply c2cwsgiutils/scripts/genversion.py:15]: good eyes !

**Comment by arnaud-morvan** [comment]
Runned `make` locally and get this error:
```
Step 13/21 : RUN python3 -m pip install --disable-pip-version-check --no-cache-dir --no-deps         --editable=/opt/c2cwsgiutils &&     python3 -m compileall -q &&     python3 -m compileall /usr/local/lib/python3.8 /usr/lib/python3.8 /opt/c2cwsgiutils -q &&     python3 -c 'import c2cwsgiutils'
 ---> Running in 0134f514c9d6
Obtaining file:///opt/c2cwsgiutils
Installing collected packages: c2cwsgiutils
  Running setup.py develop for c2cwsgiutils
Successfully installed c2cwsgiutils
Can't list '/usr/lib/python38.zip'
*** Error compiling '/usr/local/lib/python3.8/dist-packages/pipenv/patched/yaml2/constructor.py'...
  File "/usr/local/lib/python3.8/dist-packages/pipenv/patched/yaml2/constructor.py", line 167
    except TypeError, exc:
                    ^
SyntaxError: invalid syntax

*** Error compiling '/usr/local/lib/python3.8/dist-packages/pipenv/patched/yaml2/reader.py'...
  File "/usr/local/lib/python3.8/dist-packages/pipenv/patched/yaml2/reader.py", line 163
    except UnicodeDecodeError, exc:
                             ^
SyntaxError: invalid syntax

*** Error compiling '/usr/local/lib/python3.8/dist-packages/pipenv/patched/yaml2/resolver.py'...
  File "/usr/local/lib/python3.8/dist-packages/pipenv/patched/yaml2/resolver.py", line 174
    re.compile(ur'''^(?:yes|Yes|YES|no|No|NO
                    |true|True|TRUE|false|False|FALSE
                    |on|On|ON|off|Off|OFF)$''', re.X),
                 ^
SyntaxError: invalid syntax

*** Error compiling '/usr/local/lib/python3.8/dist-packages/pipenv/patched/yaml2/scanner.py'...
  File "/usr/local/lib/python3.8/dist-packages/pipenv/patched/yaml2/scanner.py", line 1421
    except UnicodeDecodeError, exc:
                             ^
SyntaxError: invalid syntax

The command '/bin/sh -c python3 -m pip install --disable-pip-version-check --no-cache-dir --no-deps         --editable=/opt/c2cwsgiutils &&     python3 -m compileall -q &&     python3 -m compileall /usr/local/lib/python3.8 /usr/lib/python3.8 /opt/c2cwsgiutils -q &&     python3 -c 'import c2cwsgiutils'' returned a non-zero code: 1
Makefile:51: recipe for target 'build_docker' failed
make: *** [build_docker] Error 1
```

**Comment by sbrunner** [comment]
@arnaud-morvan I just pin pipenv , then your local build should work...

**Comment by arnaud-morvan** [comment]
All tests successfully passed locally.
Thanks @sbrunner

**Pull request merged**